src/ffi_utils.rs

```diff
@@ -0,0 +1,49 @@
+//! Cross platform FFI helpers.
+
+use std::ffi::CStr;
+
+// The system property named 'persist.sys.timezone' contains the name of the
+// current timezone.
+//
+// From https://android.googlesource.com/platform/bionic/+/gingerbread-release/libc/docs/OVERVIEW.TXT#79:
+//
+// > The name of the current timezone is taken from the TZ environment variable,
+// > if defined. Otherwise, the system property named 'persist.sys.timezone' is
+// > checked instead.
+const ANDROID_TIMEZONE_PROPERTY_NAME: &[u8] = b"persist.sys.timezone\0";
+
+/// Return a [`CStr`] to access the timezone from an Android system properties
+/// environment.
+pub fn android_timezone_property_name() -> &'static CStr {
```

Comment:
Potential optimization: the function signature can be made `const` if the runtime check branch inside it is removed. Since the tests are below, this seems like an easy win. That said, the compiler probably figures this out anyway. Moving the unsafe to explicitly compile-time also seems safer and could then be mentioned in the `SAFETY` docstring.

I think the function should be made `pub(crate)` to signal unambiguously this is not public API.

Reply:
The CStr function is only const on recent Rust (1.59 or something, I'm on mobile and not looking at the docs)

Reply:
Ah, I had missed that. You are right, we can't switch to `const` until we can use rust 1.59.

I think `pub(crate)` should be good, though.

Reply:
done